buf/registry/module/v1/file_descriptor_set_service.proto

```diff
@@ -0,0 +1,104 @@
+  //
+  // Once the resource is resolved, the following content is returned:
+  //   - If a Module is referenced, the FileDescriptorSet for the Commit of the default Label is returned.
+  //   - If a Label is referenced, the FileDescriptorSet for the Commit of this Label is returned.
+  //   - If a Commit is referenced, the FileDescriptorSet for this Commit is returned.
+  ResourceRef resource_ref = 1 [(buf.validate.field).required = true];
+  // Exclude imports from the returned FileDescriptorSet.
+  //
+  // A file is an import if:
+  //
+  //   - It comes from a dependency of the Module specified via the ResourceRef used in the request.
+  //     For example, if the ResourceRef specifies "buf.build/foo/bar", which depends on
+  //     "buf.build/foo/baz", all files from "buf.build/foo/baz" will be imports.
+  //   - It is a Well-Known Type, and the Well-Known Type was not part of the Module specified
+  //     via the ResourceRef used in the request. Well-Known Types are special in the Protobuf ecosystem,
+  //     and can be imported without being contained in a specified module.
+  //   - It contains types that are referenced by types specified in the request. For example,
+  //     if the request specified symbol "foo.v1.Foo" contained within "foo.proto", which has a message
+  //     field of type "baz.v1.Baz", which is contained within "baz.proto", then the file "baz.proto"
+  //     will be marked as an import, regardless of which Module it derived from. If the types field
+  //     is not specified on the request, this situation will never occur.
+  //
+  // Note that if imports are excluded, the returned FileDescriptorSet may not (and likely will not)
+  // be self-contained. If imports are included, all types referenced within the FileDescriptorSet
+  // are contained within the returned FileDescriptorSet.
+  bool exclude_imports = 2;
+  // Include SourceCodeInfo in the returned FileDescriptorProtos.
+  //
+  // SourceCodeInfo is optional additional information that is not required to describe a set
+  // of Protobuf APIs.
+  bool include_source_code_info = 3;
+  // Include source-retention options in the returned FileDescriptorProtos.
+  //
+  // With the new Editions work, options are either runtime-retention or source-retention.
+  // Setting this option will include the latter.
+  bool include_source_retention_options = 4;
+  // Only include specific types (and types that they reference, if exclude_imports is not set).
+  //
+  // A symbol is a package, message, enum, service, method, or extension.
+  //
+  // Types are referenced by their fully-qualified name. For example, if "Foo" is a message within
+  // the package "foo.v1", the fully-qualified name is "foo.v1.Foo". Fully-qualfied names should
+  // not start with a period, that is specify "foo.v1.Foo", not ".foo.v1.Foo".
+  //
+  // If no types are provided, all types are returned.
+  repeated string types = 5;
```

Comment:
Should we have a protovalidate rule for this that fqn's do not start with `.`? (Same goes for `exclude_types` if we add it).

Reply:
Go for it